src/main/java/de/tum/in/www1/artemis/web/rest/LearningPathResource.java

```diff
@@ -0,0 +1,126 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import javax.ws.rs.BadRequestException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastInstructor;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.LearningPathService;
+import de.tum.in.www1.artemis.service.feature.Feature;
+import de.tum.in.www1.artemis.service.feature.FeatureToggle;
+import de.tum.in.www1.artemis.web.rest.dto.PageableSearchDTO;
+import de.tum.in.www1.artemis.web.rest.dto.SearchResultPageDTO;
+import de.tum.in.www1.artemis.web.rest.dto.competency.LearningPathHealthDTO;
+import de.tum.in.www1.artemis.web.rest.dto.competency.LearningPathPageableSearchDTO;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+
+@RestController
+@RequestMapping("/api")
+public class LearningPathResource {
+
+    private final Logger log = LoggerFactory.getLogger(LearningPathResource.class);
+
+    private final CourseRepository courseRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final LearningPathService learningPathService;
+
+    public LearningPathResource(CourseRepository courseRepository, AuthorizationCheckService authorizationCheckService, LearningPathService learningPathService) {
+        this.courseRepository = courseRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.learningPathService = learningPathService;
+    }
+
+    /**
+     * PUT /courses/:courseId/learning-paths/enable : Enables and generates learning paths for the course
+     *
+     * @param courseId the id of the course for which the learning paths should be enabled
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @PutMapping("/courses/{courseId}/learning-paths/enable")
+    @FeatureToggle(Feature.LearningPaths)
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Void> enableLearningPathsForCourse(@PathVariable Long courseId) {
```

Comment:
personal preference; feel free to ignore: the path variables should be primitive types because the method crashes anyway if the course id is null.